File: RetrofitGetJson/app/src/main/java/com/example/retrofitgetjson/ApiInterface.java
package com.example.retrofitgetjson;

import java.util.List;

import okhttp3.RequestBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;

public interface ApiInterface {

    @GET("AppStoreSystem/files/project/arra_new/1_Authentication/ASaleLogin/login.json")
    Call<GetApi> getPost();

    @POST("posts")
    Call<Post> createUser(@Body Post post);
}

File: app/src/main/java/ApiInterface.kt

import base.BaseDataModel
import home.model.ProductModel
import login.model.UserModel
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.POST

interface ApiInterface {

//    @GET("AppStoreSystem/files/project/arra_new/1_Authentication/ASaleLogin/login.json")
//    fun getUser(): Call<UserModel>

    @GET("AppStoreSystem/files/project/svoa_online/new/2_Home/ProductList/product.json")
    fun getProduct(): Call<ProductModel>

    @Headers("Content-Type: application/json")
    @POST("AppStoreSystem/files/project/arra_new/1_Authentication/ASaleLogin/login.json")
    fun post(@Body userLogin: BaseDataModel): Call<UserModel> /*Observable<UserModel>*/

}
